src/parser.coffee

```diff
@@ -95,6 +95,42 @@ class Parser
     curTrig = null     # Pointer to the current trigger in the ast.topics
     lastcmd = ""       # Last command code
     isThat  = null     # Is a %Previous trigger
+    cmdTpl  = {
+      'define': 0,
+      'label': 0,
+      'trigger': 0,
+      'afterMatch': 0,
+      'response': 0,
+      'condition': 0,
+      'previous': 0,
+      'continue': 0,
+      'redirect': 0
+    }
+    stats   = {
+      # How many total lines
+      lines: 0
+      # How many lines of actual code
+      , codelines: 0
```

Comment:
Honestly not super sure about the necessarily of counting the actual lines to this kind of granularity. `lines` makes sense, but `codelines`, `objectlines`, `comments` less so

Do you think total size in bytes of the script would be useful as well?

Reply:
The line counters seem pretty low overhead, but if you don't think they're useful, I'll axe 'em.  Total size might actually be an expensive operation - you'd have to JSON.stringify the final AST, which might be non-trivial for something quartz size.

Reply:
Yeah, stringifying at the end isn't great, but don't you already get the actual text of each line already? Would adding `line.length` counter each time be too expensive?